unity-renderer/Assets/DCLPlugins/ECS7/Systems/AudioSourceSystem/ECSAudioSourceSystem.cs

```diff
@@ -0,0 +1,53 @@
+﻿using DCL.ECS7;
+using DCL.ECS7.ComponentWrapper.Generic;
+using DCL.ECS7.InternalComponents;
+using DCL.ECSComponents;
+using DCL.ECSRuntime;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace ECSSystems.AudioSourceSystem
+{
+    public class ECSAudioSourceSystem
+    {
+        private readonly IInternalECSComponent<InternalAudioSource> audioSourceComponent;
+        private readonly IReadOnlyDictionary<int, ComponentWriter> componentsWriter;
+        private readonly WrappedComponentPool<IWrappedComponent<PBAudioSource>> audioSourcePool;
+
+        public ECSAudioSourceSystem(
+            IInternalECSComponent<InternalAudioSource> audioSourceComponent,
+            IReadOnlyDictionary<int, ComponentWriter> componentsWriter,
+            WrappedComponentPool<IWrappedComponent<PBAudioSource>> audioSourcePool)
+        {
+            this.audioSourceComponent = audioSourceComponent;
+            this.componentsWriter = componentsWriter;
+            this.audioSourcePool = audioSourcePool;
+        }
+
+        public void Update()
+        {
+            var audioSourceComponentList = audioSourceComponent.GetForAll();
+
+            // Loop through every audio source component
+            foreach (var component in audioSourceComponentList)
```

Comment:
I think it might be worth it to switch from a `foreach` to a `for` loop to reduce allocations. What do you think?

Reply:
@AlejandroAlvarezMelucciDCL you are right! The IEnumerator call will allocate, good spot